packages/xo-web/src/common/selectors.js

```diff
@@ -589,32 +589,48 @@ export const getLoneSnapshots = create(
   )
 )
 
+const _getResolvedResourceSets = (resourceSets, networks, srs, vms) =>
+  map(resourceSets, resourceSet => {
+    const { objects, ...attrs } = resourceSet
+    const objectsByType = {}
+    const objectsFound = []
+
+    const resolve = (type, _objects) => {
+      const resolvedObjects = pick(_objects, objects)
+      if (!isEmpty(resolvedObjects)) {
+        objectsFound.push(...Object.keys(resolvedObjects))
+        objectsByType[type] = Object.values(resolvedObjects)
+      }
+    }
+    resolve('VM-template', vms)
+    resolve('SR', srs)
+    resolve('network', networks)
+
+    return {
+      ...attrs,
+      missingObjects: difference(objectsFound, objects),
+      objectsByType,
+    }
+  })
+
+export const getResolvedResourceSet = create(
+  create(
+    (_, props) => props.resourceSet,
+    _createCollectionWrapper(resourceSet =>
+      resourceSet === undefined ? undefined : [resourceSet]
+    )
+  ),
```

Comment:
Why don't you do it the other way around?

- `_getResolvedResourceSet`: function that resolves 1 resource set
- `getResolvedResourceSet`: selector that resolves 1 resource set
- `getResolvedResourceSets`: selector that resolves multiple resource sets: `map(resourceSets, _getResolvedResourceSet)`

I think it makes more sense and you wouldn't even have to use `_createCollectionWrapper`.